api/src/main/java/org/openmrs/api/db/hibernate/HibernateLocationDAO.java

```diff
@@ -128,48 +139,61 @@ public LocationTag saveLocationTag(LocationTag tag) {
 	 */
 	@Override
 	public LocationTag getLocationTag(Integer locationTagId) {
-		return (LocationTag) sessionFactory.getCurrentSession().get(LocationTag.class, locationTagId);
+		return sessionFactory.getCurrentSession().get(LocationTag.class, locationTagId);
 	}
-	
+
 	/**
 	 * @see org.openmrs.api.db.LocationDAO#getLocationTagByName(java.lang.String)
 	 */
 	@Override
-	@SuppressWarnings("unchecked")
 	public LocationTag getLocationTagByName(String tag) {
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(LocationTag.class).add(
-		    Restrictions.eq("name", tag));
-		
-		List<LocationTag> tags = criteria.list();
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<LocationTag> cq = cb.createQuery(LocationTag.class);
+		Root<LocationTag> root = cq.from(LocationTag.class);
+
+		cq.select(root).where(cb.equal(root.get("name"), tag));
```

Comment:
Could there be a reason why you did not just do `cq.where(cb.equal(root.get("name"), tag))`?

Reply:
I think it was an oversight as it is redundant, reverted this and in another place